handsontable/src/shortcuts/context.js

```diff
@@ -0,0 +1,189 @@
+import { createUniqueMap } from '../utils/dataStructures/uniqueMap';
+import { normalizeKeys } from './utils';
+import { isUndefined, isDefined } from '../helpers/mixed';
+import { isFunction } from '../helpers/function';
+import { objectEach } from '../helpers/object';
+
+/**
+ * Create shortcuts' context.
+ *
+ * @param {string} name Context's name.
+ * @returns {object}
+ */
+export const createContext = (name) => {
+  const SHORTCUTS = createUniqueMap({
+    errorIdExists: keys => `The passed keys combination "${keys}" is already registered in the "${name}" context.`
+  });
+
+  /**
+   * Add shortcut to the context.
+   *
+   * @param {object} options Options for shortcut's variants.
+   * @param {Array<Array<string>>} options.variants Shortcut's variants.
+   * @param {Function} options.callback The callback.
+   * @param {object} options.namespace Namespace for shortcut.
+   * @param {object} [options.runAction]  Option determine whether assigned callback should be performed.
+   * @param {object} [options.stopPropagation=true] Option determine whether to stop event's propagation.
+   * @param {object} [options.preventDefault=true] Option determine whether to prevent default behavior.
+   * @param {object} [options.relativeToNamespace] Namespace name, relative which the shortcut is placed.
+   * @param {object} [options.position='after'] Position where shortcut is placed. It may be added before or after
+   * another namespace.
+   *
+   */
+  const addShortcut = (
```

Comment:
It have been looking like that in first implementation. However, there are 3 required parameters, so the alternative is to have 4 parameters. Not so much, but it may be very long method execution when current `variant` property has more values. What's more it's more easy to throw objects to `addShortcuts` method. I would stay with that.